### PR TITLE
Reset password email read only

### LINKED ISSUE #1808: Reset Password Form: pre populate with user email and read only
the field email in the reset password form, must be populated with user info email and must be read only

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1808

https://user-images.githubusercontent.com/10047945/207076158-8d1cc95a-34b2-447b-924c-126ec8b3eaa1.mov

https://user-images.githubusercontent.com/10047945/207076288-c7ff4167-2c6a-4e75-897e-96454779447b.mov

Screenshot 2022-12-12 alle 18 37 04

### CODE REVIEW

**Comment by sorja** [comment]
does backend check for email === resetPasswordEmail ?

File: src/server/api/user/getResetPassword.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { Requests } from '@server/utils'
+
+export const getResetPassword = async (
+  req: Request<never, never, never, { resetPasswordUuid: string }>,
+  res: Response
+) => {
+  try {
+    const { resetPasswordUuid } = req.query
+
+    const { user, userResetPassword } = await UserController.readByResetPassword({ resetPasswordUuid })
```
Comment (minotogna) [inline src/server/api/user/getResetPassword.ts]: call this UserController.findByResetPassword

File: src/server/controller/user/readByResetPassword.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  client: BaseProtocol = DB
+): Promise<{
+  user: User
+  userResetPassword: UserResetPassword
+}> => {
+  const { resetPasswordUuid } = props
+
+  const userResetPassword = await UserResetPasswordRepository.read({ uuid: resetPasswordUuid }, client)
+
+  const user = await UserRepository.getOne({ id: userResetPassword.userId }, client)
```
Comment (minotogna) [inline src/server/controller/user/readByResetPassword.ts]: do we check some sort of validity of resetPasswordUuid?

File: src/server/api/user/getResetPassword.ts (hunk 1 above)
Comment (rfontanarosa) [reply src/server/api/user/getResetPassword.ts]: ok, I'll rename readByInvitation too

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**